lib/req/steps.ex

```diff
@@ -277,6 +277,45 @@ defmodule Req.Steps do
     end)
   end
 
+  @doc """
+  Sets the json format for Google's BigQuery API request body.
+
+  ## Request Options
+
+    * `:big_query` - sets the query to be sent to Google's BigQuery API
+
+    * `:dataset` - sets the Google's BigQuery dataset
+
+    * `:project_id` - sets the Google's Console API project id.
+
+  ## Examples
+
+      iex> Req.new(url: "https://bigquery.api/queries", big_query: "SELECT * FROM iris", dataset: "my-dataset", project_id: "my-project-id").body
```

Comment:
instead of being part of Req itself, the integration will most likely be in a req_bigquery plugin, but I'm happy to iterate on the API here.

For the sake of discussion, could you paste a code snippet how you'd use Req to make the API call today? You can skip the auth token part.

Reply:
```elixir
Mix.install([
  {:req, github: "aleDsz/req", branch: "feature/big_query_step"}
])

%Req.Response{status: 200, body: %{"rows" => rows, "schema" => %{"fields" => fields}}} =
  Req.post!("https://bigquery.googleapis.com/bigquery/v2/projects/leafy-tenure-347719/queries",
    auth: {:bearer, "token..."},
    big_query: "SELECT * FROM iris",
    dataset: "livebook",
    project_id: "leafy-tenure-347719"
  )
 ```

Reply:
I'm also trying to setup the `Req.new/1` to make it more easy and less verbose

Reply:
Thanks for the example. It uses your fork of Req though. Could you show a code sample that uses req main branch? It is just to set the baseline.

Reply:
Sure

Reply:
oh and if that is easily accessible, please also leave a link to official google docs for this particular endpoint.

Reply:
```elixir
Mix.install([
  {:req, github: "wojtekmach/req"}
])

json = ~s|{"defaultDataset":{"datasetId":"livebook","projectId":"leafy-tenure-347719"},"query":"SELECT * FROM iris"}|

%Req.Response{status: 200, body: %{"rows" => rows, "schema" => %{"fields" => fields}}} =
  Req.post!("https://bigquery.googleapis.com/bigquery/v2/projects/leafy-tenure-347719/queries", auth: {:bearer, "token"
}, body: json)
```